src/ui.h

```diff
@@ -208,6 +211,8 @@ class uilist: public ui_container
         uilist( const std::string &hotkeys_override );
         // query() will be called at the end of these convenience constructors
         uilist( const std::string &msg, const std::vector<uilist_entry> &opts );
+        uilist( const std::string &msg, const std::vector<uilist_entry_pair_t> &opts,
+                const std::function<bool( uilist_entry_pair_t )> &f );
```

Comment:
```suggestion
                const std::function<bool( uilist_entry_pair_t )> &predicate );
```

Reply:
Remove this ctor per your comment about the interface.